packages/client/src/api.ts

```diff
@@ -360,21 +374,52 @@ export class ApiClient extends BaseClient {
    * Getting data about challenge.
    * The POST method starts checking on the ACME server side.
    * @param url Challenge URI
-   * @param method Method selector
+   * @param method Method selector. Default is POST-as-GET
+   * @deprecated Should be removed
    */
-  public async getChallenge(url: string, method: "POST" | "POST-as-GET" = "POST-as-GET") {
-    const res = await this.fetch<protocol.Challenge>(url, {
-      method,
-      kid: this.getAccountId(),
-      nonce: this.nonce,
-      key: this.accountKey.privateKey,
-      body: method === "POST" ? {} : undefined,
-      convert: (resp) => resp.json(),
-    });
-    if (method === "POST") {
-      await this.pause(2000);
+  public async getChallenge(url: string, method?: "POST" | "POST-as-GET"): Promise<ApiResponse<protocol.Challenge>>;
+  /**
+   * Gets a challenge
+   * @param url Challenge URL
+   * @param params Parameters
+   */
+  public async getChallenge(url: string, params?: protocol.ChallengeGetParams): Promise<ApiResponse<protocol.Challenge>>;
```

Comment:
For some feature types of challenges we need to give parameters. Previous implementation doesn't allow to do it. I extended the method.
```js
client.getChalleng(url); // POST-as-GET retquest
client.getChalleng(url, {}); // POST retquest with params
```